src/components/CharacterList.js

```diff
@@ -1,16 +1,46 @@
 import React, { useEffect, useState } from "react";
+import axios from "axios";
+import CharacterCard from "./CharacterCard";
+import { Container, Row, Col} from "reactstrap";
 
-export default function CharacterList() {
-  // TODO: Add useState to track data from useEffect
-
+const CharacterList = () => {
+  const [list, setList] = useState([]);
   useEffect(() => {
-    // TODO: Add API Request here - must run in `useEffect`
-    //  Important: verify the 2nd `useEffect` parameter: the dependancies array!
+    const getList = () => {
+      axios
+        .get("https://rickandmortyapi.com/api/character/")
+        .then(res => {
+          const list = res.data.results;
+          console.log("Character Data", res);
+          setList(list);
+        })
+        .catch(err => {
+          console.log("Sorry, an error has occured", err);
+        });
+    };
+    getList();
   }, []);
 
   return (
-    <section className="character-list">
-      <h2>TODO: `array.map()` over your state here!</h2>
-    </section>
+    <div className="character-list">
+      <Container>
+      <Row>
+      <Col>
+      {list.map(character => {
+        return(
+          <CharacterCard 
+          key={character.id}
+          image={character.image}
+          name={character.name}
+          species={character.species}
+          />
+        );
+      })}
+      </Col>
+      </Row>
+      </Container>
+    </div>
```

Comment:
Proper Indentation:
```
<div className="character-list">
  <Container>
    <Row>
      <Col>
        {list.map(character => {
          return (
            <CharacterCard
              key={character.id}
              image={character.image}
              name={character.name}
              species={character.species}
            />
          );
        })}
      </Col>
    </Row>
  </Container>
</div>
```
**Tip:** _Anything between an opening and closting tag `<Col>` `</Col>` is indented one level._